src/api.c

```diff
@@ -591,8 +592,10 @@ static AVFrame *pop_frame(struct nmd_ctx *s)
 
         if (s->st_timebase.den) {
             int ret = nmdi_async_pop_frame(s->actx, &frame);
-            if (ret < 0)
+            if (ret < 0) {
                 TRACE(s, "poped a message raising %s", av_err2str(ret));
+                s->eof = (ret == AVERROR_EOF);
```

Comment:
Is this really enough? If there is a trim duration specified, I think the filtering mod will early exit with an `AVERROR_EXIT`

Reply:
Good catch, I double checked and added AVERROR_EXIT.